operator/controllers/lifecycle/keptntask/job_runner_builder.go

```diff
@@ -0,0 +1,49 @@
+package keptntask
+
+import (
+	"github.com/go-logr/logr"
+	klcv1alpha3 "github.com/keptn/lifecycle-toolkit/operator/apis/lifecycle/v1alpha3"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// JobRunnerBuilder is the interface that describes the operations needed to help build job specs of a task
+type JobRunnerBuilder interface {
+	// CreateContainerWithVolumes returns a job container and volumes based on the task definition spec
+	CreateContainerWithVolumes(ctx context.Context) (*corev1.Container, []corev1.Volume, error)
+}
+
+// BuilderOptions contains everything needed to build the current job
+type BuilderOptions struct {
+	client.Client
+	recorder record.EventRecorder
+	req      ctrl.Request
+	Log      logr.Logger
+	task     *klcv1alpha3.KeptnTask
+	taskDef  *klcv1alpha3.KeptnTaskDefinition
+}
+
+func getJobRunnerBuilder(options BuilderOptions) JobRunnerBuilder {
+	if isJSSpecDefined(options.taskDef.Spec) {
+		return newJSBuilder(options)
+	}
+	if isContainerSpecDefined(options.taskDef.Spec) {
+		return newContainerBuilder(options.taskDef)
+	}
+	return nil
+}
+
+func specExists(definition *klcv1alpha3.KeptnTaskDefinition) bool {
+	return isJSSpecDefined(definition.Spec) || isContainerSpecDefined(definition.Spec)
+}
+
+func isJSSpecDefined(spec klcv1alpha3.KeptnTaskDefinitionSpec) bool {
+	return spec.Function != nil
+}
+
+func isContainerSpecDefined(spec klcv1alpha3.KeptnTaskDefinitionSpec) bool {
+	return spec.Container != nil
+}
```

Comment:
I think these belong to the struct definition as functor receiver of `klcv1alpha3.KeptnTaskDefinition`
This way you can re-use them into the reconcile loop